src/support/circular.h

```diff
@@ -41,10 +42,28 @@ class Circular {
         std::unique_lock<std::mutex> l(m_mu);
         return bufferedLocked();
     }
+
+    std::queue<std::tuple<const T*, size_t>> tempQueue;
     bool enqueue(const T* data, size_t N) {
-        if (N > BUFFER_SIZE) {
-            throw std::runtime_error("Trying to enqueue too much data");
+        if (tempQueue.size()) {
+            // throw std::runtime_error("Trying to enqueue too much data");
+            tempQueue.push(std::make_tuple(data, N));
+            if (std::get<1>(tempQueue.front()) < BUFFER_SIZE) {
+                do {
```

Comment:
The whole point here is to *not stall* for too long (see the call to `wait_for` a bit later), and take advantage of a tight streamline of data between the components to reduce audio lag. This breaks those assumptions for no effective result.

Reply:
I will check it more. However, throwing an exception for the shortage of audio buffer is not a good idea.